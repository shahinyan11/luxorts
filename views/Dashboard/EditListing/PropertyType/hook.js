import {
  fetchListing,
  fetchListingPropertyType,
  fetchListingsBasePropertyTypes,
} from 'state/concepts/listings/actions';

// eslint-disable-next-line import/prefer-default-export
export const getInitialProps = async (ctx) => {
  ctx.store.dispatch(fetchListingsBasePropertyTypes());
  ctx.store.dispatch(fetchListing(ctx.query.listingId));
  ctx.store.dispatch(fetchListingPropertyType(ctx.query.listingId));
};
