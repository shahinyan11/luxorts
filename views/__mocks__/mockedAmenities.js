export default {
  'Amina Heaney': {
    '5e066ad7-4194-4bd4-95ec-0d9715f8159b': {
      id: '5e066ad7-4194-4bd4-95ec-0d9715f8159b',
      description: 'Maiores porro nostrum dolorum sunt.',
      name: 'Nolan Kertzmann',
      status: 'active',
      category: null,
      checked: false,
    },
  },
  'Torrie Sawayn': {
    'c277033f-1683-4d2d-a409-d00b0bbe8201': {
      id: 'c277033f-1683-4d2d-a409-d00b0bbe8201',
      description: 'Nemo magnam dolores quam voluptatibus distinctio unde.',
      name: 'Stephine Rohan',
      status: 'active',
      category: null,
      checked: false,
    },
  },
};
