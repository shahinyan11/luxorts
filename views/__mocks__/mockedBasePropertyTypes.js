export default [
  {
    value: '51e35121-c27e-46fc-9feb-b51dab2490ee',
    label: 'repudiandae',
    description: 'description repudiandae',
  },
  {
    value: '995c5714-185d-4085-8512-690903334f6e',
    label: 'molestias',
    description: 'description molestias',
  },
];
