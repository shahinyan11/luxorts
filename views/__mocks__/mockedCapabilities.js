export default {
  APPROVED: {
    '11da7909-9f4c-45d6-a2a0-8a105b067ed2': {
      id: '11da7909-9f4c-45d6-a2a0-8a105b067ed2',
      description: 'Vel commodi ad incidunt repellendus a molestias delectus nam.',
      name: 'Tonya Wunsch',
      status: 'active',
      checked: false,
    },
    'b84c898e-2176-4ac8-9dfa-54186ccef6cb': {
      id: 'b84c898e-2176-4ac8-9dfa-54186ccef6cb',
      description: 'Maxime nostrum ex iste velit nulla incidunt.',
      name: 'Edwin Greenholt',
      status: 'active',
      checked: false,
    },
  },
};
