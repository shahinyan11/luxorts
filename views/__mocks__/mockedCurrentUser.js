export default {
  id: '5c5d814c-8e06-433d-b006-4119311585bd',
  email: 'evhenii.hryhoriev+10@rubygarage.org',
  publicId: 'c4toxHVi',
  verifiedStatus: 'unverified',
  userType: 'traveller_account',
  userProfile: {
    id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
    about: null,
    address: null,
    avatarUrls: {
      original: '/avatar-img-url',
    },
    dateOfBirth: '1994-01-01',
    drivingLicenseFirstName: 'Evhenii',
    drivingLicenseLastName: 'Test',
    firstName: 'Evhenii',
    gender: null,
    lastName: 'Test',
    phoneNumber: '+299999999',
    street: 'street',
    city: 'city',
    state: 'state',
    zipCode: 'zipCode',
    country: 'country',
  },
  userInvitation: [
    {
      id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
      created_at: '2022-06-23T06:32:55.588Z',
      email: '12@mail.ru',
      last_sent_at: '2022-06-23T06:32:55.474Z',
      resendable: true,
      status: 'pending',
      user_account_id: '7211ccc2-1fca-4799-b1e1-4005e9fc42be',
    },
  ],
};
