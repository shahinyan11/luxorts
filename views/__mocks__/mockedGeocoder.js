export default [
  {
    streetNumber: '240',
    street: '240 North 7th Avenue',
    city: 'Phoenix',
    state: 'Arizona',
    country: 'United States',
    zipCode: '85007',
    fullAddress: '240 N 7th Ave, Phoenix, AZ 85007, USA',
    longitude: -112.0825682,
    latitude: 33.4507026,
  },
];
