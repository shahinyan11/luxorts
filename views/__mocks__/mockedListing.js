export default {
  id: '0645baf3-945d-4122-8a0b-5ab3e95a9b3e',
  applicationStep: 'accomodation',
  status: 'in_draft',
  userAccount: { id: '6cbf2b68-4a26-41b5-bcd5-161cc37e1420', type: 'hostAccount' },
  listingPropertyType: {
    id: 'ea19f127-bc89-4799-bdbb-346126ac3845',
    typeOfPlace: 'private_place',
    propertyType: {
      id: '51e35121-c27e-46fc-9feb-b51dab2490ee',
      name: 'repudiandae',
      description: 'description repudiandae',
    },
  },
  listingLocation: {
    id: '2b4032ec-a469-4489-b7dc-536d8f9552eb',
    country: 'United States',
    city: 'Phoenix',
    state: 'Arizona',
    street: 'West Maryland Avenue',
    apartmentNumber: '12',
    zipCode: '12345',
    longitude: '-73.97447621784085',
    latitude: '40.764468117826794',
  },
  listingDescription: null,
  listingPricing: {
    pricePerDay: 64,
  },
};
