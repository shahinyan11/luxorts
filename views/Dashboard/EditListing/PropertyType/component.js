import { Skeleton } from 'antd';

import useListingPropertyType from 'hooks/listing/useListingPropertyType';

import EditListingLayout from 'views/layouts/EditListing';
import EditListingFooter from 'views/shared/footers/EditListing';
import ListingPropertyType from 'views/shared/listing/PropertyType';

import { getInitialProps } from './hook';

const PropertyType = () => {
  const { formik, propertyTypesOptions, showSkeleton, isLoading, tooltip } = useListingPropertyType(
    {
      shouldRedirect: false,
      message: { description: { id: 'shared.changesHaveBeenSaved' } },
    },
  );

  if (showSkeleton) {
    return (
      <div className="new-listing__container mb-24">
        <Skeleton active />
      </div>
    );
  }

  return (
    <>
      <ListingPropertyType
        formik={formik}
        propertyTypesOptions={propertyTypesOptions}
        tooltip={tooltip}
      />
      <EditListingFooter
        disabled={showSkeleton}
        isLoading={isLoading}
        onSave={formik.handleSubmit}
      />
    </>
  );
};

PropertyType.Layout = EditListingLayout;

PropertyType.getInitialProps = getInitialProps;

export default PropertyType;
