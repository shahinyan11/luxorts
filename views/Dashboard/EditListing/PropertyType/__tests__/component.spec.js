import { shallow } from 'enzyme';

import useListingPropertyType from 'hooks/listing/useListingPropertyType';

import { formik } from '__mocks__/formik';
import mockedBasePropertyTypes from 'views/__mocks__/mockedBasePropertyTypes';

import PropertyType from '../component';

const mockedHookData = {
  formik,
  propertyTypesOptions: mockedBasePropertyTypes,
  tooltip: 'tooltip',
  showSkeleton: false,
};
jest.mock('hooks/listing/useListingPropertyType', () => jest.fn(() => mockedHookData));

describe('PropertyType component', () => {
  let component = shallow(<PropertyType />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when showSkeleton equals true', () => {
    useListingPropertyType.mockReturnValueOnce({
      ...mockedHookData,
      showSkeleton: true,
    });

    component = shallow(<PropertyType />);

    expect(component).toMatchSnapshot();
  });
});
