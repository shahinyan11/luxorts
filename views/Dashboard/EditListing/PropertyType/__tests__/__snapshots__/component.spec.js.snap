// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyType component matches snapshot 1`] = `
<Fragment>
  <ListingPropertyType
    formik={
      Object {
        "dirty": true,
        "handleSubmit": [MockFunction],
        "isValid": true,
        "resetForm": [MockFunction],
        "setFieldTouched": [MockFunction],
        "setFieldValue": [MockFunction],
        "setValues": [MockFunction],
        "validateForm": [MockFunction],
        "values": Object {},
      }
    }
    propertyTypesOptions={
      Array [
        Object {
          "description": "description repudiandae",
          "label": "repudiandae",
          "value": "51e35121-c27e-46fc-9feb-b51dab2490ee",
        },
        Object {
          "description": "description molestias",
          "label": "molestias",
          "value": "995c5714-185d-4085-8512-690903334f6e",
        },
      ]
    }
    tooltip="tooltip"
  />
  <EditListingFooter
    disabled={false}
    isLoading={false}
    onSave={[MockFunction]}
  />
</Fragment>
`;

exports[`PropertyType component matches snapshot when showSkeleton equals true 1`] = `
<div
  className="new-listing__container mb-24"
>
  <Skeleton
    active={true}
    avatar={false}
    paragraph={true}
    title={true}
  />
</div>
`;
